src/sentry/integrations/perforce/client.py

```diff
@@ -226,7 +229,47 @@ def get_depots(self) -> list[dict[str, Any]]:
         finally:
             self._disconnect(p4)
 
-    def get_user(self, username: str) -> dict[str, Any] | None:
+    def get_author_info(
+        self,
+        username: str,
+        user_cache: dict[str, dict[str, Any] | None],
+        p4: Any | None = None,
+    ) -> tuple[str, str]:
+        """
+        Get author name and email from Perforce, using cache to avoid duplicate lookups.
+
+        Args:
+            username: Perforce username
+            user_cache: Dictionary cache for user lookups
+            p4: Optional P4 connection to reuse (avoids creating new connection)
+
+        Returns:
+            Tuple of (author_email, author_name)
+        """
+        # Default values
+        author_email = f"{username}@perforce"
+        author_name = username
+
+        # Fetch user info if not in cache (skip "unknown" placeholder)
+        if username != "unknown" and username not in user_cache:
```

Comment:
nit: I think we should assume people pass in a username they actually want to look up
```suggestion
        if username not in user_cache:
```